### PR TITLE
Log WMF api warnings so we can catch and fix them.

### PR DESCRIPTION
**Pull request by montehurd**
I noticed a few different API warnings when playing with various feed requests. Figured we could catch these more easily in the future if we logged warnings.

### CODE REVIEW

File: Wikipedia/Code/WMFApiJsonResponseSerializer.m [hunk 1]
```diff
@@ -27,6 +27,14 @@ - (id)responseObjectForResponse:(NSURLResponse *)response
         }
         return nil;
     }
+
+#if DEBUG
+    NSDictionary *warnings = json[@"warnings"];
+    if (warnings) {
+        NSLog(@"\n\n\nWarnings for request:\n%@\n%@", response.URL.absoluteString, warnings);
```
Comment (montehurd) [inline Wikipedia/Code/WMFApiJsonResponseSerializer.m]: I suppose we could also assert that there are no warnings... would that be preferred?

Comment (joewalsh) [reply Wikipedia/Code/WMFApiJsonResponseSerializer.m]: I think I'd rather just log for now - I think there were a few cases of expected warnings in places search. Also use `DDLogWarn` for consistency (even though it's wrapped in #if DEBUG)

Comment (montehurd) [reply Wikipedia/Code/WMFApiJsonResponseSerializer.m]: Ah will do.

Comment (montehurd) [reply Wikipedia/Code/WMFApiJsonResponseSerializer.m]: Fixed!

**Pull request merged**